### PR TITLE
glogger: Fix Python 3.8 compatibility

### PR DESCRIPTION
**Pull request by Jongy**
I thought mypy would've caught it

### CODE REVIEW

File: glogger/extra_adapter.py [hunk 1]
```diff
@@ -34,11 +34,11 @@ def process(self, msg: Any, kwargs: MutableMapping[str, Any]) -> Tuple[Any, Muta
                 other_kwargs[k] = v
 
         # Merge other kwargs into extra:
-        extra = logging_kwargs.get("extra", {}) | other_kwargs
+        extra: Mapping[str, object] = {**logging_kwargs.get("extra", {}), **other_kwargs}
```
Comment (d3dave) [inline glogger/extra_adapter.py:37]: `object`? shouldn't it be Any?

File: glogger/extra_adapter.py [hunk 2]
```diff
@@ -34,11 +34,11 @@ def process(self, msg: Any, kwargs: MutableMapping[str, Any]) -> Tuple[Any, Muta
...
         if extra:
             logging_kwargs["extra"] = extra
 
         extra = self.get_extra(**logging_kwargs)
         # Retain all extras as attributes on the record, and add "extra" attribute that contains all the extras:
-        logging_kwargs.update({"extra": extra | {"extra": extra}})
+        logging_kwargs.update({"extra": {**extra, **{"extra": extra}}})
```
Comment (d3dave) [inline glogger/extra_adapter.py:43]:
```suggestion
        logging_kwargs.update({"extra": {**extra, "extra": extra}})
```

**Review by d3dave — CHANGES_REQUESTED** [review]

File: glogger/extra_adapter.py (hunk 1 above)
Comment (Jongy) [reply glogger/extra_adapter.py:37]: Hmm I copied from `extra: Mapping[str, object]` in the c'tor, and later we have `Mapping[str, Any]`, I suppose ALL need to be `Mapping[str, Any]` right?

**Comment by Jongy** [comment]
Tests pass locally for me. Is it flakiness?
```
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.10.6, pytest-6.2.4, py-1.9.0, pluggy-0.13.1
rootdir: /.../gprofiler/granulate-utils
plugins: timeout-1.4.1, html-2.1.1, ddtrace-0.55.1, reraise-2.1.1, metadata-1.10.0
collected 14 items                                                                                                                                                                           

tests/glogger/test_extra_adapter.py ...                                                                                                                                                [ 21%]
tests/glogger/test_handler.py ..........                                                                                                                                               [ 92%]
tests/granulate_utils/test_cgroups.py .                                                                                                                                                [100%]

==================================================================================== 14 passed in 12.36s =====================================================================================

```

**Comment by Jongy** [comment]
I re-ran and the test passed so I guess it's flakiness

**Review by d3dave — APPROVED** [review]

**Pull request merged**